pkg/apis/core/types.go

```diff
@@ -5653,6 +5660,29 @@ type TopologySpreadConstraint struct {
 	// in their corresponding topology domain.
 	// +optional
 	LabelSelector *metav1.LabelSelector
+	// MinDomains indicates a minimum number of eligible domains.
+	// When the number of eligible domains with matching topology keys is less than `minDomains`,
+	// Pod Topology Spread treats "global minimum" as 0, and then the calculation of Skew is performed.
+	// As a result, when the number of eligible domains is less than `minDomains`,
+	// scheduler won't schedule more than maxSkew Pods to those domains.
+	// The default value is 1. Valid values are integers greater than 0.
+	// When value is different from 1, WhenUnsatisfiable must be DoNotSchedule.
+	//
+	// For example, in a 3-zone cluster, MaxSkew is set to 2, MinDomains is set to 5 and pods with the same
+	// labelSelector spread as 2/2/2:
+	// +-------+-------+-------+
+	// | zone1 | zone2 | zone3 |
+	// +-------+-------+-------+
+	// |  P P  |  P P  |  P P  |
+	// +-------+-------+-------+
+	// The number of domains is less than 5(MinDomains), so "global minimum" is treated as 0.
+	// In this situation, new pod with the same labelSelector cannot be scheduled,
+	// because computed skew will be 3(3 - 0) if new Pod is scheduled to any of the three zones,
+	// it will violate MaxSkew.
+	//
+	// This is an alpha field and requires enabling MinDomainsInPodTopologySpread feature gate.
+	// +optional
+	MinDomains *int32
```

Comment:
This (internal version) should be a concrete value

```suggestion
	MinDomains int32
```

Reply:
No, it should be a pointer. If it is a non-pointer, we cannot round trip through internal without populating the field value, which we don't want to do before the field graduates.

Reply:
Good to know that. I've followed the "concrete value for versioned type and pointer for unversioned types" for a while..

So when it graduates to GA, is it optional or mandatory to make it a concret value? Also, does it relate with optional/required semantics?

Reply:
> Good to know that. I've followed the "concrete value for versioned type and pointer for unversioned types" for a while..

other way around, right? pointer for versioned type and non-pointer for unversioned types?

> So when it graduates to GA, is it optional or mandatory to make it a concret value?

if you still need to distinguish between set/unset (for defaulting or validation purposes), it should still be a pointer.

Reply:
> other way around, right? pointer for versioned type and non-pointer for unversioned types?

yes, typo...

> if you still need to distinguish between set/unset (for defaulting or validation purposes)

It sounds o be applicable to the versioned type? I have no question on the versioned type - it should be always a pointer as we need default unset to 1. My confusion was on the unversioned (internal) type: should it be non-pointer since the beginning, or updated to non-pointer when it graduates?